contracts/most/UtBase.sol

```diff
@@ -0,0 +1,228 @@
+     * @param _amount Amount of UtBase tokens to be minted.
+     *
+     */
+    function mint(
+        address payable _beneficiary,
+        uint256 _amount
+    )
+        external
+        onlyConsensusCogateway
+    {
+        _mint(_beneficiary, _amount);
+        unwrapInternal(_beneficiary, _amount);
+    }
+
+    /**
+     * @notice Unwrap converts ERC20 UtBase to base coin.
+     *
+     * @dev  This contract's base coin balance must always be greater than
+     *       unwrap amount.
+     *
+     * @param _amount Amount of ERC20 UtBase token to convert to base coin.
+     */
+    function unwrap(
+        uint256 _amount
+    )
+        external
+    {
+        unwrapInternal(msg.sender, _amount);
+    }
+
+    /**
+     * @notice Wrap converts base coin to ERC20 UtBase.
+     *
+     * @dev The ERC20 OST balance of contract should always be greater than the
+     *      received payable amount.
+     *
+     *      Function requires:
+     *          - msg.value should be non-zero
+     */
+    function wrap()
+        external
+        payable
+    {
+        uint256 amount = msg.value;
+        address account = msg.sender;
+
+        require(
+            amount > 0,
+            "Payable amount should not be zero."
+        );
+
+        assert(balances[address(this)] >= amount);
+
+        transferBalance(address(this), account, amount);
+    }
+
+    /**
+     * @notice It allows to burn UtBase tokens for an account. It calls
+     *         ERC20Token::_burn method.
+     *
+     * @param _account The account whose tokens will be burnt.
+     * @param _value The amount that will be burnt.
+     */
+    function burn(address _account, uint256 _value)
+        external
+    {
+        _burn(_account, _value);
+    }
+
+    /**
+     * @notice It allows to burn tokens of the spender. It calls
+     *         ERC20Token::_burnFrom method.
+     *
+     * @param _account The account whose tokens will be burnt.
+     * @param _value The amount that will be burnt.
+     */
+    function burnFrom(address _account, uint256 _value)
+        external
+    {
+        _burnFrom(_account, _value);
+    }
+
+
+    /* Public Functions */
+
+    /**
+     * @notice Gets the consensus cogateway contract address.
+     *
+     * @return Consensus cogateway contract address.
+     */
+    function getConsensusCogateway() public view returns(address) {
+        return consensusCogateway;
+    }
+
+
+    /* Internal Functions */
+
+    /**
+     * @notice Internal method to transfer the base coin equivalent amount to
+     *         beneficiary address.
+     *
+     * @dev Function requires:
+     *          - Amount must be less than or equal to beneficiary token balance
+     *          - Caller UtBase token balance should be greater than amount
+     *
+     * @param _beneficiary Beneficairy address which will receive base coin.
+     * @param _amount Amount of ERC20 UtBase token to be unwrapped.
```

Comment:
As per the latest developer guidelines, please use pre and post conditions.

Reply:
Done.